### PR TITLE
Add search hints functionality, create SearchHint Component, add User…

### PR DESCRIPTION
**Pull request by Polo11121**
…Hints to separate functionality, add test to check if input works

# Related issue
- Search

# Changes
- Add search hints functionality, create SearchHint Component, separate functionality, user see suggestions by typed pattern, user can click suggestion and he will be redirected to clicked user profile.

# Additional information
- I've added one simple test to see if the input works. 
I'm still not sure if the functionality is well separated

# Checklist
- [X] Performed self-review
- [X] Added data-testId for UI elements / add one to check if input works correctly, hard to add because it depends on the database
- [ ] Tests / hard to add because it depends on the database
- [X] Storybook 

# Screenshots
- image
- image

### CODE REVIEW

File: src/components/Search/Search.tsx [hunk 1]
```diff
@@ -1,23 +1,51 @@
...
-    <SearchIcon className="search__icon" />
-    <input data-testid={`${testid}Input`} placeholder="Search..." className="search__input" type="text" />
-  </div>
-);
+export const Search: React.FC<SearchProps> = ({ testid, hintsFunction }) => {
+  const [searchInput, setSearchInput] = useState('');
+  const [searchHints, setSearchHints] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (hintsFunction) {
```
Comment (MateuszLisowski) [inline src/components/Search/Search.tsx:17]: maybe `getHints`? or something liek that

File: src/components/Search/Search.tsx [hunk 2]
```diff
@@ -1,23 +1,51 @@
...
+    }
+  }, [searchInput]);
+
+  return (
+    <div className="search" data-testid={testid}>
+      <div className="search__container">
+        <SearchIcon className="search__icon" />
+        <input
+          value={searchInput}
+          onChange={e => setSearchInput(e.target.value)}
```
Comment (MateuszLisowski) [inline src/components/Search/Search.tsx:28]:
```suggestion
          onChange={event => setSearchInput(e.target.value)}
```

File: src/stories/SearchHint/SearchHint.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  decorators: [
+    (Story: React.FC) => (
+      <MemoryRouter>
+        <Story />
+      </MemoryRouter>
+    )
+  ]
+};
+
+const Users = ['Jakub Gość', 'Mateusz Szklarz', 'Mateusz DJ', 'Matuesz Opiekun'];
```
Comment (MateuszLisowski) [inline src/stories/SearchHint/SearchHint.stories.tsx:16]: xD

**Review by MateuszLisowski — DISMISSED** [review]
Generally looks fine!

File: src/components/Search/Search.tsx (hunk 2 above)
Comment (Polo11121) [reply src/components/Search/Search.tsx:28]: same "e" is a bad practice?

File: src/components/Search/Search.tsx (hunk 1 above)
Comment (Polo11121) [reply src/components/Search/Search.tsx:17]: sounds good to me

File: src/components/Search/Search.tsx (hunk 2 above)
Comment (MateuszLisowski) [reply src/components/Search/Search.tsx]: I think it's better to use descriptive consts

**Review by MateuszLisowski — APPROVED** [review]
LGBT!

**Review by jakub-gosciniak — APPROVED** [review]

**Review by mateusz0612 — APPROVED** [review]

**Review by MatYouKy — APPROVED** [review]

**Pull request merged**